src/object/cli_shard.c

```diff
@@ -1102,6 +1147,21 @@ dc_rw_cb(tse_task_t *task, void *arg)
 
 	if (ret == 0 || obj_retry_error(rc))
 		ret = rc;
+	dc_obj_metrics_incr_completecntr(opc, ret);
+	if (ret == 0) {
+		daos_size_t size;
+		int is_part_stripe;
+		struct daos_oclass_attr *oca = &rw_args->shard_args->auxi.obj->cob_oca;
+
+		if (opc == DAOS_OBJ_RPC_UPDATE) {
+			size = dc_get_update_size(rw_args->shard_args->api_args->iods,
+					rw_args->shard_args->api_args->nr, oca);
+		} else {
+			size = dc_get_fetch_size(orwo);
+		}
+		is_part_stripe = is_ec_obj ? reasb_req->orr_part_stripe : 0;
```

Comment:
Thank you @wangdi1 for the review comments. I will move this block under if is_metrics_enabled. I had not done so initially since this global is not exported to other files.